lib/foreman_maintain/cli/upgrade_command.rb

```diff
@@ -52,12 +52,19 @@ def print_versions(target_versions)
         target_versions.sort.each { |version| puts version }
       end
 
+      def allow_self_upgrade?
+        !disable_self_upgrade? ||
+          ((respond_to?(:target_version) && !target_version.nil?) &&
+          (!target_version.end_with?('z') ||
+           target_version == 'nightly'))
+      end
```

Comment:
I think this logic is wrong. The old code would read `perform_self_upgrade unless disable_self_upgrade?` which means that unless the user passed `--disable-self-upgrade` the upgrade would happen.

Now, if they pass `--disable-self-upgrade --target-version nightly` the upgrade will happen, as the logic is now `if not disable_self_upgrade? OR target_version` (shortened)

Reply:
Corrected this, now the method `allow_self_upgrade?` respects the `disable_self_upgrade?` and behaves as the name says. And yes we always do the self-upgrade for the `list-versions` subcommand.

Reply:
and I am checking the tests now!